esmvaltool/diag_scripts/ocean3d/interpolation.py

```diff
@@ -0,0 +1,248 @@
+"""APPLICATE/TRR Ocean Diagnostics."""
+import logging
+import os
+import ESMF
+import numpy as np
+import pyresample
+from cartopy.util import add_cyclic_point
+from netCDF4 import Dataset
+
+from esmvaltool.diag_scripts.ocean3d.getdata import load_meta
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def closest_depth(depths, depth):
+    """Find closest depth.
```

Comment:
The preprocessor contains a function `extract_level`. What is the advantage of the present function?